New set of DebOps scripts

drybjed opened the pull request.

drybjed committed:
Initialize a new codebase for DebOps scripts

drybjed committed:
Prepare monorepo for use as an Ansible Collection

drybjed committed:
Switch 'setup.py' to use new set of DebOps scripts

ypid committed:
Change DebOps tools status to alpha for now

ypid committed:
Switch to debops-devel ML for author contact instead of user ML

ypid reviewed (dismissed):
I went ahead and pushed small fixes to your branch in order to save time. Let me know if that works for you.

This looks good. Just some input about the interface to Ansible. I would try to keep the `run` and `check` subcommands a small wrapper around the Ansible commands, potentially passing all further arguments for the Ansible based subcommands directly without further parsing. That is just from my usage of DebOps. I just call ansible-playbook directly.

For me, the real benefit we can provide with the DebOps scripts is https://github.com/debops/debops-tools/issues/164.

drybjed commented:
> I went a head and pushed small fixes to your branch in order to save time. Let me know if that works for you.

Thanks. I'm not sure how you managed to do that, I guess write status on th `debops/debops` repository gives that access.

I actually wanted to siwtch the owner email to `debops-devel`, but decided against it. I guess that I should trust my gut feeling more. :)

> This looks good. Just some input about the interface to Ansible. I would try to keep the `run` and `check` subcommands a small wrapper around the Ansible commands, potentially passing all further arguments for the Ansible based subcommands directly without further parsing. That is just from my usage of DebOps. I just call ansible-playbook directly.

That's what I plan to do. The `debops run` would be an equivalent of the current `debops` command, and `debops check` would be the same but with `--diff --check` added automatically for convenience.

> For me, the real benefit we can provide with the DebOps scripts is [debops/debops-tools#164](https://github.com/debops/debops-tools/issues/164).

We're starting from scratch so we can make things better - for example I'd like to `git init` the project directory automatically, etc. For now I'll try to fix the Travis-CI issues before merging the branch to `master`.

drybjed committed:
Correctly open the Changelog file for reading

drybjed committed:
Fix issues with UTF-8 encoding

drybjed committed:
Don't remove Python's 'egg-info' directory

Removing that directory might break the "editable" install mode.

drybjed committed:
Provide stub version number in devel environments

tasosalvas reviewed (dismissed):
I'm making a note to update [CLI Doc](https://docs.debops.org/en/master/user-guide/scripts.html) with the new usage. I might be a while, though... still crunching like a madman on my deployment.

drybjed committed:
Better organization for "DebOps Collection" dirs

With these changes, the reach of Ansible during execution should be well
scoped inside of the 'ansible/collections/' subdirectory without
exposing other pieces of code outside of it.

drybjed committed:
Implement basic project directory structure

This patch adds the ProjectDir class which defines basic DebOps project
directory structure. If needed, the class can create the project
directory on initialization and populate it with configuration files.

drybjed committed:
Add generation of 'ansible.cfg' config file

This patch adds support for generation of the 'ansible.cfg'
configuration file in the DebOps project directories. The configuration
contents are based on a basic template, user-specific configuration is
applied on top of the basic configuration.

drybjed committed:
Ansible does not like quotes in its configuration

drybjed committed:
Enable the 'timer' callback plugin by default

drybjed committed:
Add header in generated 'ansible.cfg' config file

drybjed committed:
Change how 'ansible.cfg' options are accessed

drybjed committed:
Make '--project-dir' a global option for commands

The '--project-dir' option can be used to specify the path to the DebOps
project directory in subcommands which rely on it. This can be useful if
the 'debops' script is executed from outside of a valid DebOps project
directory.

drybjed committed:
Print a more friendly error message on unknown cmd

drybjed committed:
Implement 'run' and 'check' commands

The 'debops run' and 'debops check' commands can be used to execute
Ansible playbooks against hosts, or check any changes when they are
applied, respectively. Playbooks can be specified either as direct paths
to files with or without the '.yml' extansion, or defined as
'<namespace>.<collection/><path/to/playbook>' strings which will be
expanded by DebOps to use playbooks included in Ansible Collections.

drybjed committed:
Change the order of Ansible Collections paths

This patch should ensure that Ansible Collections installed by the user
either in the project directory or in the user's home directory take
precendence over collections included with the 'debops' Python package.

drybjed committed:
Implement extensible configuration system

drybjed committed:
Reorganize project-related commands as subcommands

drybjed committed:
Use object methods to manipulate 'ansible.cfg'

drybjed committed:
Reorganize ProjectDir object to use methods

drybjed committed:
Make 'debops project' commands more consistent

drybjed committed:
Add file with default configuration options

drybjed committed:
Move inventory functions to separate module

drybjed committed:
Implement encfs lock/unlock functionality

drybjed committed:
Don't nest project configuration too deep

Previously in the DebOps configuration structure, the current project
configuration was nested under ['projects'][name] key. This could be
used to modify specific project options from the global configuration,
however this caused the per-project configuration to require nesting
each time.

This patch changes the code so that the per-project configuration is on
the same level as global configuration. This should make the per-project
config files easier to write and understand. The global state between
DebOps project can be handled some other way in the future.

drybjed committed:
Don't add DebOps collection roles in roles_path

Ansible can find roles correctly inside of a Collection when the
'collections:' keyword is specified in a playbook.

drybjed committed:
Clean up 'ansible.cfg' header code

drybjed committed:
Disable script unit tests until new ones are made

drybjed committed:
Implement initialization of EncFS secrets

drybjed committed:
Ensure that submodule is recognized by setuptools

drybjed committed:
Update the debops scripts documentation

drybjed committed:
Update support for task, file, template overrides

drybjed committed:
Update project status in setup.py

drybjed committed:
Remove old DebOps scripts codebase

drybjed committed:
Use new DebOps subcommands in documentation

drybjed committed:
Use 'debian/bullseye64' as default Vagrant box

drybjed committed:
Update Dockerfile and switch to bullseye-slim

drybjed committed:
Make sure that lookup plugins work without DebOps
